[mahadhir247] iP

mahadhir247 opened the pull request:
## Buddy
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Buddy helps you organise your tasks! It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use :eyes:

All you need to do is,
1. download it from [here](https://github.com/mahadhir247/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

mahadhir247 committed:
Renamed chatbot to Buddy. Implemented skeletal version that greets user and exits

mahadhir247 committed:
Added echo function that repeats commands entered by user and exits when user types 'bye'

mahadhir247 committed:
Added ability to store text entered by user and displays them when user types 'list'

mahadhir247 committed:
Added ability to mark/unmark tasks

mahadhir247 committed:
Added todos,events and deadlines using inheritance. Fixed a bug where you can mark/unmark tasks using the wrong commands

mahadhir247 committed:
Modified test ui testing files to reflect changes made to java files

mahadhir247 committed:
Updated chatbot to handle errors from incorrect input. Included relevant test files to test incorrect inputs

mahadhir247 committed:
Added ability to delete tasks from list, implemented using collections. Updated test files to reflect changes made. Fixed EXPECTED.TXT, which previously included incorrect results

mahadhir247 committed:
Updated to use Java enums

mahadhir247 committed:
Add save functionality

mahadhir247 committed:
Merge branch 'branch-Level-7'

mahadhir247 committed:
Add ability to understand date and time

mahadhir247 committed:
Merge branch 'branch-Level-8'

mahadhir247 committed:
Refactor code to extract related code as classes and organize classes into suitable java packages

mahadhir247 committed:
Merge branch 'branch-A-MoreOOP' and include A-Packages requirements

mahadhir247 committed:
Merge branch 'add-gradle-support'

mahadhir247 committed:
Change main class to Buddy.java

mahadhir247 committed:
Add JUnit tests for AddCommand, DeleteCommand and MarkCommand

mahadhir247 committed:
Add ability to find task

mahadhir247 committed:
Add JavaDocs for every file

mahadhir247 committed:
Update files to follow coding standard

mahadhir247 committed:
Merge branch 'branch-A-CodingStandard'

mahadhir247 committed:
Merge branch 'branch-Level-9'

mahadhir247 committed:
Merge branch 'branch-A-JavaDoc'

mahadhir247 committed:
Update build.gradle to create jar file

tanjiajiajun reviewed (commented):
Hi, I have been tasked to do a PR review on your code. I will be reviewing on coding standard violations only

1. `execute_normalInput_success()` in AddCommandTest.java should be camelCase instead eg: `executeNormalInputSuccess()`, same goes for the other test functions there.

2. Good job making the screen logo in `Ui.java` here!
```
 String logo =
                "                     ____            _     _       \n" +
                "                    |  _ \\          | |   | |      \n" +
                "                    | |_) |_   _  __| | __| |_   _ \n" +
                "                    |  _ <| | | |/ _` |/ _` | | | |\n" +
                "                    | |_) | |_| | (_| | (_| | |_| |\n" +
                "                    |____/ \\__,_|\\__,_|\\__,_|\\__, |\n" +
                "                                              __/ |\n" +
                "                                             |___/ ";
```
But since its a variable that will not change, I recommend you changing it to `private final String LOGO`

Other than that, good job for the rest of the project, the structuring of the project was very intuitive

mahadhir247 committed:
Update the GUI using javafx and add checkstyle

mahadhir247 committed:
Merge branch 'branch-Level-10'

tanjiajiajun commented on src/test/java/command/AddCommandTest.java:
```diff
@@ -0,0 +1,33 @@
+        this.testList.addTask(new Todo("task"));
+    }
+    @Test
+    public void execute_normalInput_success() {
```
1. execute_normalInput_success() in AddCommandTest.java should be camelCase instead eg: executeNormalInputSuccess(), same goes for the other test functions there.

tanjiajiajun reviewed (commented):
Just some nits. LGTM.

NewtonKoh reviewed (commented):
LGTM! I like how you grouped the different commands into classes. Nothing much to nit. Keep up the good work!

mahadhir247 committed:
Refactor code to improve code quality

Certain classes such as Parser and Storage have long methods.

This impacts the readability of the classes.

Let's extract code fragments that can be grouped together into a
method.

This makes it easier to read and can potentially help with identifying
hidden bugs.

mahadhir247 committed:
Add assertions to various classes

Assertions are used to define assumptions about the program state at
various points.

An assertion failure helps indicate a potential bug.

Let's add assertions to Parser and certain classes in the command
package.

These classes verify assumptions about some preconditions and
postconditions.

mahadhir247 committed:
Merge pull request #1 from mahadhir247/branch-A-CodeQuality

Refactor code to improve code quality

mahadhir247 committed:
Merge branch 'master' into branch-A-Assertions

mahadhir247 committed:
Merge pull request #2 from mahadhir247/branch-A-Assertions

Add assertions to various classes

mahadhir247 committed:
Fix bug due to incorrect assertion.

mahadhir247 committed:
Add Edit command to the chatbot

Allows the user to edit and task they have added without the need to
delete and add a new task.

Let's allow users to edit specific fields without modifying other
fields.

mahadhir247 committed:
Merge branch 'branch-C-Update'

mahadhir247 committed:
Improve the GUI to make it more polished

mahadhir247 committed:
Add more JUnit tests to test new features

mahadhir247 committed:
Merge branch 'branch-A-BetterGui'

mahadhir247 committed:
Merge branch 'branch-A-MoreTesting'

mahadhir247 committed:
Update user and buddy images

mahadhir247 committed:
Add User Guide and product screenshot

mahadhir247 committed:
Update docs to include exit feature

mahadhir247 committed:
Update docs to fix GitHub Pages

mahadhir247 committed:
Update README.md
